From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 8 Sep 2025 05:44:25 +0200
Subject: [PATCH] Use standard visibility flags

Workaround for
ld.lld: error: undefined hidden symbol: __libc_single_threaded
---
 build/moz.configure/toolchain.configure | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index b2192e406678..061a12db0f8b 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -2426,13 +2426,7 @@ set_define("_LIBCPP_HIDE_FROM_ABI", libcxx_override_visibility.hide_from_abi)
 @depends(target, build_environment)
 def visibility_flags(target, env):
     if target.os != "WINNT":
-        if target.kernel in ("Darwin", "FreeBSD", "OpenBSD"):
-            return ("-fvisibility=hidden", "-fvisibility-inlines-hidden")
-        return (
-            "-I%s/system_wrappers" % os.path.join(env.dist),
-            "-include",
-            "%s/config/gcc_hidden.h" % env.topsrcdir,
-        )
+        return ("-fvisibility=hidden", "-fvisibility-inlines-hidden")
 
 
 @depends(target, visibility_flags)
